velox/benchmarks/QueryBenchmarkBase.cpp

```diff
@@ -195,6 +237,40 @@ void QueryBenchmarkBase::initialize() {
   connector::registerConnector(hiveConnector);
   parquet::registerParquetReaderFactory();
   dwrf::registerDwrfReaderFactory();
+
```

Comment:
Is it convenient for you to add a new class `CudfQueryBenchmarkBase` and extends the current class `QueryBenchmarkBase`?  Then `ifdef VELOX_ENABLE_CUDF` is reduced.

Reply:
Our aim here is to use `TpchBenchmark` which derives from `QueryBenchmarkBase`. If I add a `CudfQueryBenchmarkBase` then I'll have to repeat the code in TpchBenchmark.cpp for cudf.

Reply:
@devavret Another way is to create a test util API for cudf and add the `#ifdef VELOX_ENABLE_CUDF` calls in the test util. The `#else` part will be empty.
This way, we can keep this file clean and reuse the test util for other places like examples, etc.
We follow this pattern for registering file system adapters. The client code is clean this way.

Reply:
We may also have CudfTpchBenchmark, in case of we may add wave, spark tpch benchmark or other extensions.
`class CudfTpchBenchmark : publicTpchBenchmark, public CudfQueryBenchmarkBase`

Reply:
How about now https://github.com/facebookincubator/velox/pull/13695/files/e4915773e30d8f63b826771f68af438fc782e845..2ea6ec09725f399d7afe655752ddb7864879173b